sdcm/utils/database_query_utils.py

```diff
@@ -150,15 +151,25 @@ def validate_rows_per_partitions(self, ignore_limit_rows_number: bool = False):
             LOGGER.debug('Validate partitions info')
             partitions_dict_after = self.collect_partitions_info(ignore_limit_rows_number=ignore_limit_rows_number)
             if partitions_dict_after is not None:
-                if not ignore_limit_rows_number and self.limit_rows_number:
-                    missing_rows = {key: val for key, val in partitions_dict_after.items() if
-                                    val < self.limit_rows_number}
-                    assert not missing_rows, f"Found missing rows for partitions: {missing_rows}"
-                else:
-                    self.tester.assertEqual(self.partitions_dict_before,
-                                            partitions_dict_after,
-                                            msg='Row amount in partitions is not same before and after running of nemesis: '
-                                                f' {partitions_dict_after}')
+                try:
+                    if not ignore_limit_rows_number and self.limit_rows_number:
+                        missing_rows = {key: val for key, val in partitions_dict_after.items() if
+                                        val < self.limit_rows_number}
+                        assert not missing_rows, f"Found missing rows for partitions: {missing_rows}"
```

Comment:
what's the point of doing assert, catching, and then raising event ?

can't we just raise events as needed ?

Reply:
not sure it's more readable or maintainable. fixing anyway.